src/Symfony/Component/Validator/Constraints/NotSuspiciousValidator.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Validator\Constraints;
+
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+use Symfony\Component\Validator\Exception\UnexpectedTypeException;
+use Symfony\Component\Validator\Exception\UnexpectedValueException;
+use Symfony\Contracts\Translation\LocaleAwareInterface;
+
+/**
+ * @author Mathieu Lechat <mathieu.lechat@les-tilleuls.coop>
+ */
+class NotSuspiciousValidator extends ConstraintValidator implements LocaleAwareInterface
+{
+    private string $locale;
+
+    public function validate(mixed $value, Constraint $constraint)
+    {
+        if (!$constraint instanceof NotSuspicious) {
+            throw new UnexpectedTypeException($constraint, NotSuspicious::class);
+        }
+
+        if (null === $value || '' === $value) {
+            return;
+        }
+
+        if (!\is_scalar($value) && !$value instanceof \Stringable) {
+            throw new UnexpectedValueException($value, 'string');
+        }
+
+        $value = (string) $value;
+        if ('' === $value) {
+            return;
+        }
+
+        $checker = new \Spoofchecker();
+
+        $checker->setRestrictionLevel($constraint->restrictionLevel);
+
+        $allowedLocales = $constraint->profileLocales;
+        if ($constraint->addCurrentLocaleToProfile) {
+            $allowedLocales[] = $this->locale;
+        }
+        $checker->setAllowedLocales(implode(',', $allowedLocales));
+
+        $checker->setChecks($constraint->checks);
+
+        if (!$checker->isSuspicious($value)) {
+            return;
+        }
+
+        $this->context->buildViolation($constraint->message)
+            ->setParameter('{{ value }}', $this->formatValue($value))
+            ->addViolation()
+        ;
```

Comment:
Looks more obvious to me to have this on the if. If it's suspicious we add the violation

Reply:
Building the violation could be more complex (e.g. if `isSuspicious`’s second argument worked) so you wouldn’t want to nest all the logic inside a conditional. This is called the “early return” pattern.